### ISSUE TITLE
Analysisd cannot start when overwriting a rule with a configured active response

### ISSUE THREAD

**Issue by antoniomanuelfr** [wazuh/wazuh#13576]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.3.x | Analysisd | Manager | Packages/Sources | Linux |

Hello team!

We have discovered a corner case that is causing Analysisd to not start properly. This problem can be reproduced using a Wazuh 4.3.x. 
## Actual behavior
The steps to reproduce the problem are:
- Create the following rule file `/var/ossec/etc/rules/0270-web_appsec_rules_edit.xml` with the following content:
```xml
<group name="web,appsec,attack,">
  <rule id="31533" level="9" frequency="50" timeframe="5" overwrite="yes">
    <if_matched_sid>31530</if_matched_sid>
    <same_source_ip/>
    <description>Local - High amount of POST requests in a small period of time (likely bot) (50/5s).</description>
    <group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
</group>
```

- Add an active response:
```xml
<active-response>
  <disabled>no</disabled>
  <command>firewall-drop</command>
  <location>defined-agent</location>
  <agent_id>001</agent_id>
  <rules_id>550</rules_id>
  <level>10</level>
  <timeout>1</timeout>
</active-response>
```

If we try to start the Wazuh manager, analysisd is not going to start.
## Reason for the fail
Analysisd loads the rules in alphabetic order. The failure occurs while binding between the rule and the active response that is triggered is done before checking if there is a rule that will overwrite the previous read rule. When the rule is overwritten, the AR field is freed and leaving the AR list with a corrupt memory. This is the cause of the problem. Is important that in order to reproduce the failure, make sure that the file with the rule isn't read at the end. 

### Output
```
Starting Wazuh...
Job for wazuh-manager.service failed because the control process exited with error code.
See "systemctl status wazuh-manager.service" and "journalctl -xe" for details.

root@ubuntumanager:/home/vagrant/wazuh/src# wazuh-control info
WAZUH_VERSION="v4.3.1"
WAZUH_REVISION="40311"
WAZUH_TYPE="server"
root@ubuntumanager:/home/vagrant# /var/ossec/bin/wazuh-analysisd -t
Segmentation fault
```
## Expected behavior
Analysisd should properly bind the AR fields with the overwrite flag.

**Comment by khangcnttspkt** [comment]
Hi @antoniomanuelfr,
I was posted this problem on slack, after write rule inside `0270-web_appsec_rules_edit.xml` to `local_rules.xml`, I was tested with `/var/ossec/bin/wazuh-analysisd -t` this seem ok. But then I have one new problem.

I was write to `local_rules.xml` with rules:
`
<group name="web,appsec,attack,">
  <rule id="31533" level="9" frequency="50" timeframe="5" overwrite="yes">
    <if_matched_sid>31530</if_matched_sid>
    <same_source_ip/>
    <description>Local - High amount of POST requests in a small period of time (likely bot) (50/5s).</description>
    <group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
  <rule id="800007" level="10">
    <if_matched_sid>31533</if_matched_sid>
    <list field="srcip" lookup="not_address_match_key">etc/lists/web-attacked-white-list</list>
    <description>Local - High amount of POST requests in a small period of time (likely bot) (50/5s).</description>
    <group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
    <info>Excluded White-list</info>
  </rule>
</group>


<group name="web,accesslog,">
  <rule id="31151" level="9" frequency="50" timeframe="60" overwrite="yes">
    <if_matched_sid>31101</if_matched_sid>
    <same_source_ip/>
    <description>Local - Multiple web server 400 error codes (50/1m) from same source ip.</description>
    <group>web_scan,recon,pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
  <rule id="800001" level="10">
    <if_matched_sid>31151</if_matched_sid>
    <list field="srcip" lookup="not_address_match_key">etc/lists/web-attacked-white-list</list>
    <description>Local - Multiple web server 400 error codes (50/1m) from same source ip.</description>
    <group>web_scan,recon,pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
    <info>Excluded White-list</info>
  </rule>
</group>
`
I have error
`[root@PYS-HIDS rules]# /var/ossec/bin/wazuh-analysisd -t
Segmentation fault
[root@PYS-HIDS rules]# vi local_rules.xml
[root@PYS-HIDS rules]# /var/ossec/bin/wazuh-analysisd -t
[root@PYS-HIDS rules]# vi local_rules.xml
[root@PYS-HIDS rules]# /var/ossec/bin/wazuh-analysisd -t
Bus error`

**Comment by antoniomanuelfr** [comment]
Hello @khangcnttspkt 

We are working to fix this problem. I would like to check if the `bus error` problem has been fixed. Could you share with me the exact steps that you are taking to see this? 

Thanks

**Comment by khangcnttspkt** [comment]
> Hello @khangcnttspkt
> 
> We are working to fix this problem. I would like to check if the `bus error` problem has been fixed. Could you share with me the exact steps that you are taking to see this?
> 
> Thanks

Hi @antoniomanuelfr,

When problem was fixed, next version?

Thanks.

**Comment by antoniomanuelfr** [comment]
Hello @khangcnttspkt-

The PR has been merged and the fix will be available in 4.3.2 https://github.com/wazuh/wazuh/pull/13604.

I'm closing this issue.

**Comment by khangcnttspkt** [comment]
I upgrade wazuh version 4.3.3 but cannot fix. Can you reopen this ticket?

Thanks you.

**Comment by antoniomanuelfr** [comment]
Hello @khangcnttspkt 

Unfortunately, 4.3 (among others) was affected by an issue with the NVD that causes `wazuh-modulesd` to crash. Due to this issue, the 4.3.3 version was released but without this change. This is the PR that fixed the problem https://github.com/wazuh/wazuh/pull/13624 and the 4.3.3 changelog https://github.com/wazuh/wazuh/blob/4.3/CHANGELOG.md

4.3.4 will definitely have this problem patched. Sorry for any inconvenience.

**Comment by antoniomanuelfr** [comment]
Hello @khangcnttspkt 


Unfortunately, this fix will be available in 4.3.4, as the 4.3.3 version fixes a problem with vulnerability detector that was causing problems and a new version was released to fix this problem. 

You can check the changelog for 4.3.3 and for 4.3.4 at this link https://github.com/wazuh/wazuh/issues/13576. You will see that the PR that fixes the problem (https://github.com/wazuh/wazuh/pull/13604) will be added in 4.3.4
